adafruit_requests.py

```diff
@@ -37,22 +37,51 @@
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_Requests.git"
 
 import errno
+from types import TracebackType
+
+try:
+    from typing import Union, TypeVar, Optional, Dict, Any, List, Type
+    import types
+    import ssl
+    import adafruit_esp32spi.adafruit_esp32spi_socket as esp32_socket
+    import adafruit_wiznet5k.adafruit_wiznet5k_socket as wiznet_socket
+    import adafruit_fona.adafruit_fona_socket as cellular_socket
+    from adafruit_esp32spi.adafruit_esp32spi import ESP_SPIcontrol
+    from adafruit_wiznet5k.adafruit_wiznet5k import WIZNET5K
+    from adafruit_fona.adafruit_fona import FONA
+    import socket as cpython_socket
+
+    SocketType = TypeVar(
+        "SocketType",
+        esp32_socket.socket,
+        wiznet_socket.socket,
+        cellular_socket.socket,
+        cpython_socket.socket,
+    )
+    SocketpoolModuleType = types.ModuleType
+    SSLContextType = (
+        ssl.SSLContext
+    )  # Can use either CircuitPython or CPython ssl module
+    InterfaceType = TypeVar("InterfaceType", ESP_SPIcontrol, WIZNET5K, FONA)
+
+except ImportError:
+    pass
 
 # CircuitPython 6.0 does not have the bytearray.split method.
 # This function emulates buf.split(needle)[0], which is the functionality
 # required.
-def _buffer_split0(buf, needle):
+def _buffer_split0(buf, needle: bytes):
```

Comment:
I think `bytes` is unnecessarily narrow here. You could use `_typing.ReadableBuffer`. Or maybe it is only `Union[bytes, bytearray]` ?

Reply:
I've had issues with importing `_typing` if I remember correctly, but also `Union[bytes, bytearray]` seemed more like what it was intended for